add-on/src/lib/ipfs-proxy/pre-mfs-scope.js

```diff
@@ -0,0 +1,101 @@
+const Path = require('path')
+const DEFAULT_ROOT_PATH = '/dapps'
+
+// Creates a "pre" function that is called prior to calling a real function
+// on the IPFS instance. It modifies the arguments to MFS functions to scope
+// file access to a directory designated to the web page
+function createPreMfsScope (fnName, getScope, getIpfs, rootPath = DEFAULT_ROOT_PATH) {
+  return MfsPre[fnName] ? MfsPre[fnName](getScope, getIpfs, rootPath) : null
+}
+
+module.exports = createPreMfsScope
+
+const MfsPre = {
```

Comment:
Question; how will it behave when a new `ipfs.files.foo` is added ? 
Will it not work, or will it work with no path scoping?
If the latter is the case,  we should explicitly block executions of new methods  in scoped `ipfs.files` API to avoid surprises in future.

Reply:
Yeah, I mean, you'd need to update the `ipfs-postmsg-proxy` dependency to allow the new method to be used in the first place, but no it wouldn't be scoped automatically unless you did some work here.

Reply:
Indeed, being dependent on `ipfs-postmsg-proxy` makes this safe enough for now.
Unless its trivial to add, I propose we add additional safeguards in separate PR.